Releases: Add Bitcoin Core 0.10.1

harding opened the pull request:
Preview: http://dg4.dtrt.org/en/download and http://dg4.dtrt.org/en/release/v0.10.1

This updates the download links to point to 0.10.1 and adds the 0.10.1RC3 release notes to the site.  I'll keep it updated througout the RC cycle, but it can be merged by anyone with commit access once 0.10.1 final is released (see TODO lists below).

CC: @laanwj

Essential TODO:

* [X] Figure out where the files go as discussed in #748 (discussed: it should be https://bitcoin.org/bin/bitcoin-core-0.10.1/)
* [ ] Make sure the download links work.  This is now automatically checked as part of the Travis CI build, so trigger a rebuild and, if it passes, this should be safe to merge.


Optional TODO (may be done in commits after merge):

* [ ] Add the actual release date to the YAML header in `_releases/0.10.1.md`
* [ ] Add the magnet URI to the YAML header in `_releases/0.10.1.md` (brief instructions for creating the link are provided as comments in that file)

Expected URLs for the binaries and torrent are:

    /bin/bitcoin-core-0.10.1/bitcoin-0.10.1-linux32.tar.gz
    /bin/bitcoin-core-0.10.1/bitcoin-0.10.1-linux64.tar.gz
    /bin/bitcoin-core-0.10.1/bitcoin-0.10.1-osx64.tar.gz
    /bin/bitcoin-core-0.10.1/bitcoin-0.10.1-osx.dmg
    /bin/bitcoin-core-0.10.1/bitcoin-0.10.1.tar.gz
    /bin/bitcoin-core-0.10.1/bitcoin-0.10.1.torrent
    /bin/bitcoin-core-0.10.1/bitcoin-0.10.1-win32-setup.exe
    /bin/bitcoin-core-0.10.1/bitcoin-0.10.1-win32.zip
    /bin/bitcoin-core-0.10.1/bitcoin-0.10.1-win64-setup.exe
    /bin/bitcoin-core-0.10.1/bitcoin-0.10.1-win64.zip
    /bin/bitcoin-core-0.10.1/SHA256SUMS.asc

(Torrent should be created automatically within 5 minutes of files being uploaded.)

(Travis rebuild instructions, in case anyone needs them: at the bottom of this PR where it says Travis failed, click "Details" to go to the failed build on Travis.org; on the right side on the screen beneath the Settings button, click the semicircle-arrow icon to rebuild.)

harding commented:
Updated:

* c726309 updates the release notes to match upstream (including changing the download link per discussion)

* 5561165 adds a Javascript change from #805 that would trigger a notice from the monitoring script.  Since updating the binary URLs will trigger its own notice, I'm combining the changes.

laanwj commented:
I'm always in doubt what should be the correct filenames, especially for the -osx dmg. Would these match the page?
```
bitcoin-0.10.1-linux32.tar.gz
bitcoin-0.10.1-linux64.tar.gz
bitcoin-0.10.1-osx.dmg
bitcoin-0.10.1-osx64.tar.gz
bitcoin-0.10.1-win32-setup.exe
bitcoin-0.10.1-win32.zip
bitcoin-0.10.1-win64-setup.exe
bitcoin-0.10.1-win64.zip
bitcoin-0.10.1.tar.gz
```

harding commented:
@laanwj confirmed, those are what we expect.  I'll document that better today so there's no confusion in the future.  Thanks for checking!

harding commented:
Added a few commits:

* a7bc987: added instructions to our README.md describing how to prepare a release in advance, including directions on listing the expected file names in the PR.  (I've also updated the description for this pull request to do that.)  The new directions can be seen here: https://github.com/bitcoin/bitcoin.org/tree/bitcoin-core-0.10.1#preparing-a-release-in-advance

* 0c57313: updated release notes to those in RC2.

* ca97382: A layout change from pull #814 that affects the Download page Javascript, so it's being included here.

harding commented:
Rebased to fix merge conflicts.  Also squashed three of my commits.  Previews have been updated:  http://dg4.dtrt.org/en/download and http://dg4.dtrt.org/en/release/v0.10.1

harding commented:
Release notes updated to RC3.  [Preview notes](http://dg4.dtrt.org/en/release/v0.10.1) and [download page](http://dg4.dtrt.org/en/download) updated.

harding commented:
Rebased.

laanwj commented:
I kicked travis, should pass now that the files are in place

harding commented:
Merged after Travis passed.  Should be up on site within 15 minutes.  @laanwj thanks!
